docs/docs/reference/02-yaml-syntax.md

```diff
@@ -0,0 +1,638 @@
+registration `foo_components::Foo` refers to a component `Foo` in package `foo_components`.
+
+```yaml
+my_component:
+  component: foo_components::Foo
+```
+
+### Display name
+
+This optional field is identical to the [hardware interface display name](#display-name) and is used to assign a
+nicer, human-readable display name to the component when rendered as a node in the AICA interactive graph editor.
+
+### Position
+
+The `position` field is used to define the desired location of the component when rendered as a node in the AICA
+interactive graph editor. It has two subfields defining the X and Y location, respectively.
+
+This field only affects visualization of the application graph and has no other run-time effect.
+If a position is not specified, the node will be rendered at a procedurally chosen location.
+
+```yaml
+my_component:
+  position:
+    x: 100
+    y: 200
+```
+
+### Log Level
+
+The `log_level` optionally sets the log severity level for this component.
+Supported levels are: [unset, debug, info, warn, error, fatal]
+
+```yaml
+my_component:
+  log_level: debug
+```
+
+### Mapping
+
+The `mapping` field optionally defines overrides for the component name and namespace. Normally, the component node
+is instantiated with the same name as the top level component name and put on the base namespace.
+
+By specifying a mapping `name` or `namespace` or both, the instantiated node name is updated accordingly.
+
+```yaml
+# Without the mapping directive, the node name becomes /component_a
+component_a:
+  ...
+
+# With the mapping directive, the node name becomes /my_component_namespace/my_new_component_name
+component_b:
+  mapping:
+    name: my_new_component_name
+    namespace: my_component_namespace
+```
+
+### Parameters
+
+The `parameters` field allows initial parameters values to be set using a `name: value` syntax.
+Currently, only string and double parameters are supported. These values are only applied when the component
+is loaded and are not dynamically reconfigurable.
+
+```yaml
+my_component:
+  parameters:
+    my_string_parameter: "my_string_value"
+    my_double_parameter: 2.0
+```
+
+### Inputs and Outputs
+
+The `inputs` and `outputs` fields are used to connect component signals together to enable communication, signal
+processing and control loops. Each signal is specified using a `name: value` syntax, where the name is the name
+of the signal according to the component description, and the value is and the name of the signal topic.
+If a component output is assigned to the same topic name as another component input, they are connected, as
+illustrated in the example below.
+
+```yaml
+my_component:
+  inputs:
+    robot_state: "/state"
+    applied_force: "/force"
+  outputs:
+    robot_command: "/command"
+
+my_other_component:
+  outputs:
+    force_torque_sensor: "/force"
+```
+
+#### Component period
```

Comment:
This is my first comment, so let me know if I am doing it wrong. 
Suggestion: component period would do better directly below Parameters as opposed to Inputs and Outputs.